### ISSUE TITLE
Express Checkout - Revisit billing/shipping validation issues in WC 3.0..??

### ISSUE THREAD

**Issue by angelleye** [angelleye/paypal-woocommerce#711]
* https://wordpress.org/support/topic/cannot-get-to-paypal-from-cart/#post-9116392
* https://mail.google.com/mail/u/0/#inbox/15bfe1bd356fe603
* https://www.angelleye.com/support/scp/tickets.php?id=1158
* https://mail.google.com/mail/u/0/#inbox/1555f30e6ea69908
* https://wordpress.org/support/topic/save-abandoned-checkouts/
* https://www.angelleye.com/support/scp/tickets.php?id=1199 ( another ticket , sent release branch waiting for reply. )
* https://wordpress.org/support/topic/shipping-address1-empty-error-code-10727/#post-9188987
ticket for this one: https://www.angelleye.com/support/scp/tickets.php?id=1237

>I am getting this problem as well when trying to use Express Checkout.

>Working through it I think this is being caused by the fact that Woocommerce is setting a shipping/billing country for the customer as soon as they view the cart. I think this maybe down to using the GeoIP to determine customers location.

>So then when the customer clicks on the Paypal Express checkout buttons on the site, the plugin is complaining about that the Name/Address1/Town/Etc is empty because the country has been filled in.

>I can only assume the shipping/billing country being prefilled is causing the plugin to assume the form has been filled in and the required information is missing.

**Comment by kcppdevelopers** [comment]
I have checked multiple and it seems working well for me. @MansoorAliKhanOfficial  Can you please help me to generate issue?

**Comment by angelleye** [comment]
@kcppdevelopers  Are you testing both while signed in to the site as well as not?

**Comment by kcppdevelopers** [comment]
@angelleye yes

**Comment by angelleye** [comment]
@kcppdevelopers Please see additional notes left by one of the users reporting this in the URL above...

>I spoke to a few customers who are unable to use the Express checkout, and something which is common between them is they are previous customers – who have ordered before.

>After asking them to clear the woocommerce cookies / session ID stuff from their browser – the issues goes away and they are able to use Express Checkout.

>So is there some sort of conflict occuring for cookies set pre woo v3 and woo v3 & paypal for woocommerce v1.4.3?

**Comment by angelleye** [comment]
@kcppdevelopers Maybe it has something to do with different values set under WooCommerce -> Settings -> General.  The "Default customer location" option has options for Gelocation, Gelocation with cache, etc.  Seems like I remember we ran into some issues with those before..??  Based on the feedback above it seems like this could be related..??

**Comment by angelleye** [comment]
@kcppdevelopers Something is going on here because I'm getting more reports.  It seems that sometimes the WC validation gets in the way when it shouldn't be, and other times they get to PayPal but then a partial address was sent to PayPal so we get back errors like this:

>PayPal SetExpressCheckout API call failed
Detailed Error Message: The field Shipping Address1 is required
Short Error Message: Shipping Address1 Empty
Error Code: 10727
Error Severity Code: Error

**Comment by angelleye** [comment]
>Ones I spoke to no - they are not signed up as a customer but they had previously visited the site.
 
>Here is a snippet from another failed order.
 
>[SOLUTIONTYPE] => Sole
    [LANDINGPAGE] => Billing
    [USERSELECTEDFUNDINGSOURCE] => CreditCard
    [PAYMENTREQUEST_0_AMT] => 38.80
    [PAYMENTREQUEST_0_CURRENCYCODE] => EUR
    [PAYMENTREQUEST_0_PAYMENTACTION] => Sale
    [PAYMENTREQUEST_0_SHIPTONAME] =>
    [PAYMENTREQUEST_0_SHIPTOCOUNTRYCODE] => DE
 
 
>The SHIPTOCOUNTRYCODE has been filled in, and then SHIPTONAME is being sent blank - response is error.
 
>Like I said in my previous email - I can vpn into say Germany - go to the site, add-to-cart, then go to checkout - and see that the country field is Germany (due to the Geolocating), and click on the Paypal_Express button at the top of the checkout page - and I get sent to the Paypal website, no errors.

**Comment by angelleye** [comment]
Another report added to the list of URLs above:

>if I add the product to the cart, and go to checkout page, if I don’t fill anything in the billing and shipping address, and select PayPal express method from general list of gateways,then click button, it will show “10727 – The field Shipping Address1 is required”, I think this is correct, but, after this, if I click the paypal express button on the cart page, or the checkout page( not in the general list of gateways on this page), it will check the shipping address and show “10727 – The field Shipping Address1 is required”, can’t redirect to the paypal page. in 1.3.3, it can redirect to the Paypal in the same situation.

**Comment by kcppdevelopers** [comment]
@angelleye we have resolved in the issue #735 branch. Can you please ask user to if they want to grab it early from here?

**Comment by angelleye** [comment]
I have responded to each of the linked threads above to ask if people can grab the release branch and confirm this is fixed.  I'll leave it open for now until I hear some feedback.

**Comment by MansoorAliKhanOfficial** [comment]
another ticket and thread on WordPress Forum stated as below

> We’ve talked to a couple customers who were unable to complete their orders due to this Address error and we have done test orders in effort to establish cause of the address1 missing error.
> 
> Our research and tests concluded that this error is triggered when:
> 
> 1. customer enters address data in our WooCom shopping cart
> 
> 2. only when billing and shipping addresses are different
> 
> 3. only when customer clicks to pay with PayPal (rather than paying by credit card)
> 
> There are no errors when:
> 
> 1. customer clicks ‘Pay with Paypal’ option and does not enter address data in our WooCom shopping cart
> 
> 2. when billing and shipping addresses are the same
>

**Comment by angelleye** [comment]
@kcppdevelopers Lots of feedback here:  https://wordpress.org/support/topic/shipping-address1-empty-error-code-10727/#post-9188987

**Comment by angelleye** [comment]
@kcppdevelopers I just reviewed the feedback in that thread I linked above, and I think it's the same thing we just discussed on Friday about how the new review page we have in EC fights with the general flow of it because of that "ship to a different address" check box.  I can't remember which issue that was, though..??

**Comment by kcppdevelopers** [comment]
issue #735

**Comment by angelleye** [comment]
@kcppdevelopers #735 was closed 15 days ago and is referring to validation on the WC checkout page.  The user in the link above responded 2 days ago with lots of feedback about how things aren't working smoothly, and seems to be more related to the new review page and the confusion surrounding that "ship to a different address" check box option it contains which makes no sense at that point.  This is what we discussed in depth last week.

So I think this is maybe done now, but the issue number you referenced actually isn't correct...or am I missing something..??
